jax/scipy/optimize/bfgs_minimize.py

```diff
@@ -0,0 +1,127 @@
+"""The Broyden-Fletcher-Goldfarb-Shanno minimization algorithm.
+https://pages.mtu.edu/~struther/Courses/OLD/Sp2013/5630/Jorge_Nocedal_Numerical_optimization_267490.pdf
+"""
+
+import jax
+import jax.numpy as np
+from jax.lax import while_loop
+from .line_search import line_search
+import collections
+
+BFGSResults = collections.namedtuple(
+    'BFGSResults', [
+        'converged',  # bool, True if minimization converges
+        'failed',  # bool, True if line search fails
+        'k',  # The number of iterations of the BFGS update.
+        'nfev',  # The total number of objective evaluations performed.
+        'ngev',  # total number of jacobian evaluations
+        'nhev',  # total number of hessian evaluations
+        'x_k',  # A tensor containing the last argument value found during the search. If the search converged, then
+        # this value is the argmin of the objective function.
+        'f_k',  # A tensor containing the value of the objective
+        # function at the `position`. If the search
+        # converged, then this is the (local) minimum of
+        # the objective function.
+        'g_k',  # A tensor containing the gradient of the
+        # objective function at the
+        # `final_position`. If the search converged
+        # the max-norm of this tensor should be
+        # below the tolerance.
+        'H_k'  # A tensor containing the inverse of the estimated Hessian.
+    ])
+
+
+def bfgs_minimize(func, x0, options=None):
```

Comment:
Consider prefixing ``_`` if this is meant to be private. Same below.

Reply:
I'm not sure bfgs_minimize should be considered private yet.

Reply:
If this is going to be public API, it's interface needs to match `scipy.optimize.fmin_bfgs`: https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.fmin_bfgs.html

Reply:
Done.